### ISSUE TITLE
Language model training

### ISSUE THREAD

**Issue by stefan-it** [zalandoresearch/flair#105]
Hi,

I would like to train a language model (as soon as I receive my new RTX card) and I have some questions about the training procedure:

* In another issue I found out that no tokenization is required, so I guess end-of-sentence detection is also not required?
* What was the size of the training, development and test set for e.g. the German model? The original paper just mentions the OPUS source and final perplexity, but how many MB were used for training, development and test set?

Thanks many in advance :)

**Comment by alanakbik** [comment]
Hi, great! 

- Yes, no tokenization and no end-of-sentence detection is required. Just put the text in "as-is". 
- In the original work we had about half a billion tokens for German, of which we used 1 / 500 as development and 1 / 500 as test data. Current experiments use a bit more - we are now at some 800 million to 1 billion tokens per language. 

Are you looking at a particular domain of German? What corpora are you considering?

**Comment by stefan-it** [comment]
Hi Alan,

thanks for your helpful hints!

For the beginning I want to train a model for historic German - my other project is POS tagging for Macedonian, so I wanted to train a language model for that language :)

**Comment by stefan-it** [comment]
@alanakbik Another question to the training parameters. I'm currently training a lm for Bulgarian, with:

* `hidden_size=1024`
* `sequence_length=250`
* `mini_batch_size=100`
* `max_epochs=500`

What do you use for `nlayers` for your trained lms?

**Comment by alanakbik** [comment]
@stefan-it We've pretty much always used 1 layer, which works well!

We have some experiments running with more than one, but results are not in yet. If we find that 2 or more works well, I'll post it here!

**Comment by alejandrojcastaneira** [comment]
Hello, the sequence_length parameter it refers to the max sequence of each text sample? and if we If would like to use preserve the tokenization from a previous custom tokenizer would be possible, although later will be split by characters, etc?

**Comment by alanakbik** [comment]
Sequence length is the parameter that controls "truncated backpropagation through time" (BPTT) meaning how far back in time the gradients propagate. If your sequence length is 250, a gradient will propagate backwards a maximum of 250 steps, then it gets detached.
